### PR TITLE
[NEW MODEL]Add DALL-E mini Model

### PR DESCRIPTION
**Pull request by JunnYu**
### PR types

New features
### PR changes

Models
### Description

Add DALL-E mini Model

jax version：https://github.com/borisdayma/dalle-mini
pytorch version：https://github.com/kuprel/min-dalle

### generate images
```python
import paddle
from paddlenlp.transformers import DalleBartForImageGeneration, DalleBartTokenizer
from PIL import Image

paddle.set_device("gpu:5")
model_name_or_path = "./dalle-mega"
model = DalleBartForImageGeneration.from_pretrained(model_name_or_path)
tokenizer = DalleBartTokenizer.from_pretrained(model_name_or_path)
model.eval()

top_k = 50
condition_scale = 10.0
num_return_sequences = 8

prompts = [
    "New York Skyline with 'Google Research Pizza Cafe' written with fireworks on the sky.",
    "Dali painting of WALL·E"
]

tokenized_inputs = tokenizer(
    prompts,
    return_tensors="pd",
    padding="max_length",
    truncation=True,
    return_attention_mask=True,
    max_length=64,
)
images = model.generate(**tokenized_inputs,
                      top_k=top_k,
                      condition_scale=condition_scale,
                      num_return_sequences=num_return_sequences)
images = (images.cpu().numpy().clip(0, 1) * 255).astype("uint8")
images = images.transpose([0, 2, 1, 3,
                           4]).reshape(-1, images.shape[-3],
                                       num_return_sequences * images.shape[-2],
                                       images.shape[-1])
for i, image in enumerate(images):
    image = Image.fromarray(image)
    image.save(f"figure_{i}.png")
```

#### dalle-mega
figure_0
figure_1

#### dalle-mini
minfigure_0
minfigure_1

### CODE REVIEW

File: paddlenlp/transformers/dallebart/modeling.py [hunk 1]
```diff
@@ -0,0 +1,1720 @@
...
+                "specified when generating attention_mask")
+            attention_mask = (paddle.cast(
+                input_ids == self.text_pad_token_id,
+                dtype=paddle.get_default_dtype()).unsqueeze([1, 2]) * -1e4)
+        # For 2D attention_mask from tokenizer
+        elif attention_mask.ndim == 2:
+            attention_mask = paddle.unsqueeze(
+                attention_mask, axis=[1, 2]).astype(paddle.get_default_dtype())
+            attention_mask = (1.0 - attention_mask) * -1e4
+            attention_mask.stop_gradient = True
```
Comment (guoshengCS) [inline paddlenlp/transformers/dallebart/modeling.py:852]: attention mask这里后面可能还要跟进HF和这里调整下 https://github.com/PaddlePaddle/PaddleNLP/pull/2005

File: paddlenlp/transformers/dallebart/modeling.py [hunk 2]
```diff
@@ -0,0 +1,1720 @@
...
+        self.conv1 = nn.Conv2D(m, n, 3, padding=1)
+        self.norm2 = nn.GroupNorm(2**5, n)
+        self.conv2 = nn.Conv2D(n, n, 3, padding=1)
+        if not self.is_middle:
+            self.nin_shortcut = nn.Conv2D(m, n, 1)
+
+    def forward(self, x):
+        h = x
+        h = self.norm1(h)
+        h *= F.sigmoid(h)
```
Comment (guoshengCS) [inline paddlenlp/transformers/dallebart/modeling.py]: 看Dalle-mini使用的那个VQGAN的代码里好像用的激活不太一样 https://github.com/patil-suraj/vqgan-jax/blob/main/vqgan_jax/modeling_flax_vqgan.py#L127

File: paddlenlp/transformers/dallebart/tokenizer.py [hunk 3]
```diff
@@ -0,0 +1,506 @@
...
+                                                sep_token=sep_token,
+                                                cls_token=cls_token,
+                                                unk_token=unk_token,
+                                                pad_token=pad_token,
+                                                mask_token=mask_token)
+        self.normalize_text = normalize_text
+        if self.normalize_text:
+            self.text_processor = TextNormalizer(wiki_word_frequency_file)
+        super().__init__(vocab_file, merges_file, errors, max_len, pad_token,
+                         eos_token)
```
Comment (guoshengCS) [inline paddlenlp/transformers/dallebart/tokenizer.py]: 这里把unk_token也传进去吧，不然gpt默认的unk_token会把这个给覆盖，BART的tokenzier也有一样的问题

**Review by guoshengCS — DISMISSED** [review]

File: paddlenlp/transformers/dallebart/modeling.py (hunk 2 above)
Comment (JunnYu) [reply paddlenlp/transformers/dallebart/modeling.py]: swish(x) = x * sigmoid(x) 现在已经使用F.swish

File: paddlenlp/transformers/dallebart/tokenizer.py (hunk 3 above)
Comment (JunnYu) [reply paddlenlp/transformers/dallebart/tokenizer.py]: done

**Review by guoshengCS — DISMISSED** [review]
LGTM

**Review by guoshengCS — DISMISSED** [review]

**Comment by guoshengCS** [comment]
另外在想如果这个任务本身就是ForImageGeneration的话是不是也用generate好呢，现在是用的decode是吧

**Review by guoshengCS — APPROVED** [review]
